packages/yourloops/test/integration/pages/profile/patient-profile.spec.tsx

```diff
@@ -0,0 +1,145 @@
+
+  beforeAll(() => {
+    mockAuth0Hook(UserRoles.patient)
+    mockUserDataFetch({ profile, preferences, settings })
+    mockNotificationAPI()
+    mockDirectShareApi()
+    mockTeamAPI()
+    mockPatientAPI()
+  })
+
+  it('should render profile page for a french patient and be able to edit his profile', async () => {
+    const expectedProfile = { ...profile, firstName: 'Jean', lastName: 'Tanrien', fullName: 'Jean Tanrien' }
+    const expectedPreferences = { displayLanguageCode: 'en' as LanguageCodes }
+    const updateProfileMock = jest.spyOn(UserApi, 'updateProfile').mockResolvedValue(expectedProfile)
+    const updatePreferencesMock = jest.spyOn(UserApi, 'updatePreferences').mockResolvedValue(expectedPreferences)
+
+    await act(async () => {
+      renderPage('/preferences')
+    })
+    checkPatientLayout(`${profile.firstName} ${profile.lastName}`)
+    const fields = checkPatientProfilePage(settings.country)
+    const saveButton = screen.getByRole('button', { name: 'Save' })
+
+    expect(fields.firstNameInput).toHaveValue(profile.firstName)
+    expect(fields.lastNameInput).toHaveValue(profile.lastName)
+    expect(fields.birthdayInput).toHaveValue(profile.patient.birthday)
+    expect(fields.birthPlaceInput).toHaveValue(profile.patient.birthPlace)
+    expect(fields.birthFirstNameInput).toHaveValue(profile.patient.birthFirstName)
+    expect(fields.birthLastNameInput).toHaveValue(profile.patient.birthLastName)
+    expect(fields.birthNamesInput).toHaveValue(profile.patient.birthNames)
+    expect(fields.birthPlaceZipcodeInput).toHaveValue(profile.patient.birthPlaceInseeCode)
+    expect(fields.insInput).toHaveValue(profile.patient.ins)
+    expect(fields.ssnInput).toHaveValue(profile.patient.ssn)
+    expect(fields.oidInput).toHaveValue(profile.patient.oid)
+    expect(fields.unitsSelect).toHaveTextContent(settings.units.bg)
+    expect(fields.languageSelect).toHaveTextContent('Français')
+    expect(fields.genderSelect).toHaveTextContent('Male')
+    expect(fields.referringDoctorInput).toHaveValue(profile.patient.referringDoctor)
+    expect(saveButton).toBeDisabled()
+
+    fireEvent.mouseDown(within(screen.getByTestId('profile-local-selector')).getByRole('button'))
+    fireEvent.click(screen.getByRole('option', { name: 'English' }))
+
+    userEvent.clear(fields.firstNameInput)
+    userEvent.clear(fields.lastNameInput)
+    await userEvent.type(fields.firstNameInput, 'Jean')
+    await userEvent.type(fields.lastNameInput, 'Tanrien')
+
+    expect(saveButton).not.toBeDisabled()
+    await act(async () => {
+      userEvent.click(saveButton)
+    })
+
+    expect(saveButton).toBeDisabled()
+    expect(screen.getByRole('alert')).toBeVisible()
+    expect(updatePreferencesMock).toHaveBeenCalledWith(loggedInUserId, expectedPreferences)
+    expect(updateProfileMock).toHaveBeenCalledWith(loggedInUserId, expectedProfile)
+  })
+
+  it('should render profile page without specific INS fields when patient is not french', async () => {
```

Comment:
Is this test usefull? I don't see what is tested in this one that is not already tested in the one above thinking

Reply:
It's due to commented code in [packages/yourloops/test/integration/assert/profile.ts](https://github.com/mdblp/blip/pull/941/files#diff-1eb3765c70cd9054c9f2e599d90a61c9df20d33a7900b285b85813183e61dd3a)`. Currently the frProId is not setted anymore, but will be reactivated in a futur PR. It will display the CPS number if the hcp is french. Can we keep this commented code for now ?

Reply:
It check that INS and SSN fileds are not present in the page if the patient is not french. See the `checkPatientProfilePage` method lines 127 and 128